src/vm/analysis/definition_sorter/mod.rs

```diff
@@ -109,7 +116,26 @@ impl <'a> DefinitionSorter {
                                         self.probe_for_dependencies_in_tuple(&function_args[2], tle_index)?;
                                     }
                                     return Ok(());
-                                }
+                                },
+                                DefineFunctions::Trait => {
+                                    if function_args.len() != 2 {
+                                        return Ok(())
+                                    } 
+                                    if let Some(trait_sig) = function_args[1].match_list() {
+                                        for func_sig in trait_sig.iter() {
+                                            if let Some(func_sig) = func_sig.match_list() {
+                                                if func_sig.len() == 3 {
```

Comment:
What happens if `func_sig.len()` is not 3?  Should this be an error?

Reply:
I think that, conventionally (implicit, so @kantai to chime in), the type-checker is responsible for catching and throwing this kind of errors.

Reply:
Ah, okay.  If we're certain that the type checker will not allow `func_sig` to have anything other than 3 items, then I think the code should either:

* Implement a type variant for this function signature that enforces this invariant at compile-time, or
* `panic!()` if `func_sig.len()` is ever not 3.

Reply:
The type checker runs after this pass -- so it's definitely possible that this check would fail (and shouldn't panic). This should throw an argument count or invalid trait syntax error.